core/domain/question_jobs_one_off_test.py

```diff
@@ -174,3 +175,91 @@ def test_migration_job_fails_with_invalid_question(self):
                      [u'Question %s failed validation: linked_skill_ids is '
                       'either null or an empty list' % question_id]]]
         self.assertEqual(expected, [ast.literal_eval(x) for x in output])
+
+
+class QuestionSummaryModelOneOffJobTest(test_utils.GenericTestBase):
+    """Tests for QuestionSummaryModel migration."""
+
+    ONE_OFF_JOB_MANAGERS_FOR_TESTS = [
+        question_jobs_one_off.QuestionSummaryModelOneOffJob]
+
+    def _run_one_off_job(self):
+        """Runs the one-off MapReduce job."""
+        job_id = (
+            question_jobs_one_off.QuestionSummaryModelOneOffJob.create_new())
+        question_jobs_one_off.QuestionSummaryModelOneOffJob.enqueue(job_id)
+        self.assertEqual(
+            self.count_jobs_in_taskqueue(
+                taskqueue_services.QUEUE_NAME_ONE_OFF_JOBS), 1)
+        self.process_and_flush_pending_tasks()
+        stringified_output = (
+            question_jobs_one_off.QuestionSummaryModelOneOffJob.get_output(
+                job_id)
+        )
+
+        eval_output = [ast.literal_eval(stringified_item)
+                       for stringified_item in stringified_output]
+        output = [(eval_item[0], int(eval_item[1]))
+                  for eval_item in eval_output]
+        return output
+
+    def _check_model_validity(
+            self, instance_id, original_question_summary_model):
+        """Checks if the model was migrated correctly."""
+        migrated_question_summary_model = (
+            question_models.QuestionSummaryModel.get_by_id(instance_id))
+        self.assertEqual(
+            migrated_question_summary_model.creator_id,
+            original_question_summary_model.creator_id)
+        self.assertEqual(
+            migrated_question_summary_model.question_model_last_updated,
+            original_question_summary_model.question_model_last_updated)
+        self.assertEqual(
+            migrated_question_summary_model.question_model_created_on,
+            original_question_summary_model.question_model_created_on)
+        self.assertEqual(
+            migrated_question_summary_model.question_content,
+            original_question_summary_model.question_content)
+
+    def test_successful_migration(self):
+        instance_id = 'id_1'
+        question_summary_model = question_models.QuestionSummaryModel(
+            id=instance_id,
+            creator_id='user_id',
+            question_model_last_updated=datetime.datetime.utcnow(),
+            question_model_created_on=datetime.datetime.utcnow(),
+            question_content='smth'
+        )
+        question_summary_model.put()
```

Comment:
I'm not sure whether there is any way to test the expected behavior for this one of job. Like this `question_summary_model` will be created out of the new model class and the creator_id will be indexed already, right?

Reply:
Yes, I think that this job is simple so we don't need any more complicated testing.